BKMoneyKit/BKBirthdayField.m

```diff
@@ -0,0 +1,242 @@
+    }
+    
+    if ([numberOnlyString length] > 2) {
+        NSInteger numOfDayDigits = numberOnlyString.length > 3 ? 2 : 1;
+        NSString *dayString = [numberOnlyString substringWithRange:NSMakeRange(2, numOfDayDigits)];
+        
+        if (dayString.length == 1) {
+            NSInteger dayInteger = dayString.integerValue;
+            
+            if (dayInteger < 0 || dayInteger > 3) {
+                return NO;
+            }
+        } else if (dayString.length == 2) {
+            NSInteger dayInteger = dayString.integerValue;
+            
+            switch (monthInteger) {
+                case 1:
+                case 3:
+                case 5:
+                case 7:
+                case 8:
+                case 10:
+                case 12:
+                    if (dayInteger < 1 || dayInteger > 31) {
+                        return NO;
+                    }
+                    break;
+                case 4:
+                case 6:
+                case 9:
+                case 11:
+                    if (dayInteger < 1 || dayInteger > 30) {
+                        return NO;
+                    }
+                    break;
+                case 2:
+                    // Month is typed before year, won't know if it's a leap year, bound check against upper limit
+                    if (dayInteger < 1 || dayInteger > 29) {
+                        return NO;
+                    }
+                    break;
+                default:
+                    return NO;
+                    break;
+            }
+        }
+        [formattedString appendString:dayString];
+    }
+    
+    if (numberOnlyString.length > 3) {
+        [formattedString appendString:@" / "];
+    }
+    
+    if (numberOnlyString.length > 4) {
+        NSString *yearString = [numberOnlyString substringFromIndex:4];
+        [formattedString appendString:yearString];
+    }
+    
+    [self setText:formattedString];
+    
+    [self sendActionsForControlEvents:UIControlEventEditingChanged];
+    
+    return NO;
+}
+
+- (NSDateComponents *)dateComponentsWithString:(NSString *)string
+{
+    NSString *numberString = [self numberOnlyStringWithString:string];
+    
+    NSDateComponents *result = [[NSDateComponents alloc] init];
+    result.year = 0;
+    result.month = 0;
+    result.day = 0;
+    
+    if (numberString.length > 1) {
+        result.month = [[numberString substringToIndex:2] integerValue];
+    }
+    
+    if (numberString.length > 2) {
+        NSInteger numOfDayDigits = (numberString.length > 3) ? 2 : 1;
+        result.day = [[numberString substringWithRange:NSMakeRange(2, numOfDayDigits)] integerValue];
+    }
+    
+    if (numberString.length > 4) {
+        result.year = [[numberString substringFromIndex:4] integerValue];
+    }
+    
+    return result;
+}
+
++ (BOOL)isLeapYearForYear:(NSInteger)year {
+    
+    if (year % 4 == 0 && (year % 100 != 0 || year % 400 == 0)) {
```

Comment:
@bkook 

NSCalendar has a method for checking if the current month is a leap month [NSCalendar leapMonth], but not a leap year.